src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -70,9 +70,12 @@ static int smart_amp_init(struct processing_module *mod)
 	}
 
 	/* Copy the pin formats */
-	bs = sizeof(sad->ipc4_cfg.input_pins) + sizeof(sad->ipc4_cfg.output_pin);
-	memcpy_s(sad->ipc4_cfg.input_pins, bs,
-		 base_cfg->base_cfg_ext.pin_formats, bs);
+	in_size = sizeof(sad->ipc4_cfg.input_pins);
+	out_size = sizeof(sad->ipc4_cfg.output_pin);
+	memcpy_s(sad->ipc4_cfg.input_pins, in_size,
+		 base_cfg->base_cfg_ext.pin_formats, in_size);
+	memcpy_s(&sad->ipc4_cfg.output_pin, out_size,
+		 base_cfg->base_cfg_ext.pin_formats + in_size, out_size);
```

Comment:
Pointer arithmetic on a flexible array member, wow! I wonder what that means. It looks suspicious... maybe this instead?

```suggestion
		 &base_cfg->base_cfg_ext.pin_formats[0] + in_size, out_size);
```

Reply:
> Pointer arithmetic on a flexible array member, wow! I wonder what that means.

Most likely the same thing as a regular array, sorry the noise. In that case:

```suggestion
		 &base_cfg->base_cfg_ext.pin_formats[in_size], out_size);
```

Reply:
this may cause memory leak, it may add pointer based on pin_format size instead of byte, better continue add (char *) at beginning in case xtensa compiler is not so strong.
(char *)&base_cfg->base_cfg_ext.pin_formats[0] + in_size

or 
(char *)&base_cfg->base_cfg_ext.pin_formats[1] ?

Reply:
>  in case xtensa compiler is not so strong.

The xtensa compiler is based on either GCC or Clang so there's no standard compliance concern.